hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -251,12 +251,14 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     } catch (NoSuchObjectException e) {
       throw new NoSuchTableException("Table does not exist: %s", from);
 
-    } catch (AlreadyExistsException e) {
-      throw new org.apache.iceberg.exceptions.AlreadyExistsException(
-          "Table already exists: %s", to);
-
     } catch (TException e) {
-      throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+      if (e.getMessage() != null
+          && e.getMessage().contains(String.format("new table %s already exists", to))) {
+        throw new org.apache.iceberg.exceptions.AlreadyExistsException(
+            "Table already exists: %s", to);
+      } else {
+        throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+      }
```

Comment:
could we please move this under a `catch (InvalidOperationException)`?
But keep the old `TException` handler part